### PR TITLE
Added one if Statement to Check for Optional Value

### PR DESCRIPTION
**Pull request by Mr-Anyone**
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [X] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [X] **Remove all commented out code**
- [X] **Remove extra print statements**: for example, those just used for testing
- [X] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

Fix forgot to check an condition

### CODE REVIEW

File: src/software/sensor_fusion/sensor_fusion.cpp [hunk 1]
```diff
@@ -198,17 +196,22 @@ bool SensorFusion::shouldTrustRobotStatus()
...
-    // robot, and the distance between what SSL says and where the robots are actually
-    // at is less than a threshold distance set by
-    // DISTANCE_THRESHOLD_FOR_BREAKBEAM_FAULT_DETECTION
-    return distance <= DISTANCE_THRESHOLD_FOR_BREAKBEAM_FAULT_DETECTION;
+    // In other words, we trust the breakbeam reading from robot status if vision also
+    // agrees that the ball is roughly near the robot. If vision has the ball far from the
+    // breakbeam detection, then we will ignore the breakbeam detection and trust vision
+    // instead.
+    return distance(robot_with_ball_in_dribbler->position(), ball->position()) <=
+           DISTANCE_THRESHOLD_FOR_BREAKBEAM_FAULT_DETECTION;
```
Comment (nimazareian) [inline src/software/sensor_fusion/sensor_fusion.cpp:211]: you need to import distance

**Review by williamckha — DISMISSED** [review]

File: src/software/sensor_fusion/sensor_fusion.cpp [hunk 2]
```diff
@@ -198,17 +196,22 @@ bool SensorFusion::shouldTrustRobotStatus()
...
     }
 
-    double distance =
-        (robot_with_ball_in_dribbler.value().position() - ball.value().position())
-            .length();
+    // Check if vision detects a ball on the field
+    if (!ball.has_value())
+    {
+        return false;
+    }
```
Comment (nimazareian) [inline src/software/sensor_fusion/sensor_fusion.cpp]: Shouldn't we trust robot status if ball isn't visible by vision?

Comment (Mr-Anyone) [reply src/software/sensor_fusion/sensor_fusion.cpp]: Actually, that could work.

File: src/software/sensor_fusion/sensor_fusion.cpp (hunk 1 above)
Comment (Mr-Anyone) [reply src/software/sensor_fusion/sensor_fusion.cpp:211]: yeah, sure.

File: src/software/sensor_fusion/sensor_fusion.cpp (hunk 2 above)
Comment (Mr-Anyone) [reply src/software/sensor_fusion/sensor_fusion.cpp]: Actually, that also make sense. 

Used to be that we check for all the optional value, and if we can't unwrap all the optional value, we would return false.

**Review by williamckha — APPROVED** [review]

**Review by nimazareian — APPROVED** [review]

**Pull request merged**